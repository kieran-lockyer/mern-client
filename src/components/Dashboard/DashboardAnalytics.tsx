import React from "react";
import AnalyticsCard from "./AnalyticsCard";
import StatsItem from "./StatsItem";
import styled from "styled-components";
import { connect } from "react-redux";

const DashboardAnalytics = props => {
  return (
    <>
      <AvgStats>
        <StatsItem icon="media" avgPhotos={props.avgPhotos} />
        <StatsItem icon="tag" avgTags={props.avgTags} />
      </AvgStats>
      <Container>
        <AnalyticsCard title="Popular Tags" />
        <AnalyticsCard title="Trending Tags" />
      </Container>
    </>
  );
};

const Container = styled.section`
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
`;

const AvgStats = styled.div`
  display: flex;
  justify-content: space-between;
  width: 100%;
  margin-bottom: 10px;
  @media only screen and (max-width: 800px) {
    flex-direction: column;
  }
`;

const mapStateToProps = state => ({
  avgTags: state.stats.avgTags,
  avgPhotos: state.stats.avgPhoto
});

export default connect(mapStateToProps)(DashboardAnalytics);
