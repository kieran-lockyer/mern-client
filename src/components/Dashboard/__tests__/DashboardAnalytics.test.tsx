import React from 'react';
import { shallow } from 'enzyme';
import DashboardAnalytics from '../DashboardAnalytics';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

describe('<DashboardAnalytics />', () => {
  describe('render()', () => {
    test('renders the DashboardAnalytics component', () => {
      const component = shallow(<DashboardAnalytics />);
      expect(toJson(component)).toMatchSnapshot();
    });
  });
});