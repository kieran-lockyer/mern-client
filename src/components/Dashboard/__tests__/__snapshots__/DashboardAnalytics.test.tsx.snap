// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashboardAnalytics /> render() renders the DashboardAnalytics component 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
